src/utils.ts

```diff
@@ -14,14 +14,27 @@
  * limitations under the License.
  */
 
-import { spawn } from 'child_process';
+import { spawn, SpawnOptionsWithoutStdio } from 'child_process';
 import crypto from 'crypto';
 import fs from 'fs';
 import path from 'path';
 import readline from 'readline';
 import which from 'which';
 import * as vscodeTypes from './vscodeTypes';
 
+/**
+ * Spawns a process, handling Windows .cmd/.bat files that require shell execution.
+ * Node.js 20.12.2+ doesn't allow spawning .cmd/.bat files without shell: true
+ * (see https://github.com/nodejs/node/issues/52554).
+ * Tools like mise use .cmd shims on Windows, so we need shell: true for those.
+ */
+export function spawnWithShell(executable: string, args: string[], options: SpawnOptionsWithoutStdio) {
+  const ext = path.extname(executable).toLowerCase();
+  if (process.platform === 'win32' && (ext === '.cmd' || ext === '.bat'))
+    return spawn(executable, args, { ...options, shell: true });
```

Comment:
This triggers the [Node.js deprecation warning](https://nodejs.org/api/deprecations.html#DEP0190), and will probably go away in the future. Can we avoid that?